hw3_tester.py

```diff
@@ -144,11 +144,11 @@ def verify_mdp(mdp):
             actions.append(a)
 
         if s['id'] == 0:
-            init_is_terminal = False
-            if fixed_n_actions == 1 and s['actions'][0]['transitions'][0]['to'] == 0:
-                init_is_terminal = True
-            elif s['actions'][0]['transitions'][0]['to'] == 0 and s['actions'][1]['transitions'][0]['to'] == 0:
-                init_is_terminal = True
+            init_is_terminal = True
+            for action in s['actions']:
+                for transition in action['transitions']:
+                    if transition['to'] != 0:
```

Comment:
This is related to #11. This doesn't account for the case of a less than 1.0 probability of a self-transition. Also, you need to ensure that each action has only 1 transition with probability=1.0, there could be 1 action with only a self-transition, and the other action could include other non-self transitions and the MDP shouldn't qualify for `init_is_terminal`.

Reply:
I'm not sure I follow. My logic was basically this: "if there is any action that I can take from state 0, such that any of the transitions out of that state-action takes me to some other state, then there is a way out of the start state, and therefore it isn't a terminal state." Since we're iterating over each transition of each action, if there is a way out of s0, we'll find it. If there isn't, then we raise an error. Since there are already checks implemented to ensure that the transition probabilities are in (0, 1] and that they sum to 1, checking the 'to' field should suffice. This will even handle weird edge cases like multiple transition entries to the same state.
If possible, could you perhaps attach an example mdp where this logic fails, so I can understand better?